modules/core/src/main/scala/dev/profunktor/redis4cats/effect/JRFuture.scala

```diff
@@ -16,40 +16,49 @@
 
 package dev.profunktor.redis4cats.effect
 
-import java.util.concurrent.{ CompletableFuture, CompletionStage, Future }
-
-import cats.effect.{ Async, ContextShift }
-import cats.implicits._
+import cats.Defer
+import cats.effect.{ Concurrent, ContextShift }
 import cats.effect.implicits._
+import cats.implicits._
 import io.lettuce.core.{ ConnectionFuture, RedisFuture }
+import java.util.concurrent._
 
 object JRFuture {
 
   private[redis4cats] type JFuture[A] = CompletionStage[A] with Future[A]
 
-  def apply[F[_]: Async: ContextShift, A](fa: F[RedisFuture[A]]): F[A] =
+  def apply[F[_]: Concurrent: ContextShift: Defer, A](fa: F[RedisFuture[A]]): F[A] =
     liftJFuture[F, RedisFuture[A], A](fa)
 
-  def fromConnectionFuture[F[_]: Async: ContextShift, A](fa: F[ConnectionFuture[A]]): F[A] =
+  def fromConnectionFuture[F[_]: Concurrent: ContextShift: Defer, A](fa: F[ConnectionFuture[A]]): F[A] =
     liftJFuture[F, ConnectionFuture[A], A](fa)
 
-  def fromCompletableFuture[F[_]: Async: ContextShift, A](fa: F[CompletableFuture[A]]): F[A] =
+  def fromCompletableFuture[F[_]: Concurrent: ContextShift: Defer, A](fa: F[CompletableFuture[A]]): F[A] =
```

Comment:
Defer is implied by Sync :)

Reply:
Yeah my brain was exploding last night, didn't see this one :smile: Anyway, this will disappear in the next iteration when I ditch the underlying async API.